rust/lance/src/dataset.rs

```diff
@@ -918,6 +920,17 @@ impl Dataset {
         blob::take_blobs(self, row_ids, column.as_ref()).await
     }
 
+    /// Take [BlobFile] by row indices.
+    ///
+    pub async fn take_blobs_by_indices(
```

Comment:
I'd rather this be a single method (e.g. something more like `TakeBuilder`) but we can iron out the Rust API later too.